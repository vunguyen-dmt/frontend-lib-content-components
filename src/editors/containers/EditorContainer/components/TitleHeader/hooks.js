import React from 'react';
import { useSelector } from 'react-redux';

import { actions, selectors } from '../../../../data/redux';
import * as textEditorHooks from '../../hooks';
import * as module from './hooks';

export const { navigateCallback } = textEditorHooks;

export const state = {
  // eslint-disable-next-line react-hooks/rules-of-hooks
  localTitle: (args) => React.useState(args),
};

export const hooks = {
  isEditing: () => {
  // eslint-disable-next-line react-hooks/rules-of-hooks
    const [isEditing, setIsEditing] = React.useState(false);
    return {
      isEditing,
      startEditing: () => setIsEditing(true),
      stopEditing: () => setIsEditing(false),
    };
  },

  localTitle: ({ dispatch, stopEditing }) => {
  // eslint-disable-next-line react-hooks/rules-of-hooks
    const title = useSelector(selectors.app.displayTitle);
    const [localTitle, setLocalTitle] = module.state.localTitle(title);
    return {
      updateTitle: (e) => {
        if (localTitle.length <= 0) {
          setLocalTitle(title);
          stopEditing();
        } else if (!e.currentTarget.contains(e.relatedTarget)) {
          dispatch(actions.app.setBlockTitle(localTitle));
          stopEditing();
        }
      },
      handleChange: (e) => setLocalTitle(e.target.value),
      cancelEdit: () => {
        setLocalTitle(title);
        stopEditing();
      },
      localTitle,
    };
  },
};

export const localTitleHooks = ({ dispatch }) => {
  const { isEditing, startEditing, stopEditing } = module.hooks.isEditing();
  const {
    localTitle,
    handleChange,
    updateTitle,
    cancelEdit,
  } = module.hooks.localTitle({
    dispatch,
    stopEditing,
  });
  return {
    isEditing,
    startEditing,
    stopEditing,
    cancelEdit,
    localTitle,
    updateTitle,
    handleChange,

    inputRef: React.createRef(),
  };
};
