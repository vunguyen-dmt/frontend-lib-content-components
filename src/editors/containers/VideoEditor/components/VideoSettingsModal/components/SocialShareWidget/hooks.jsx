import { useSelector } from 'react-redux';
import { sendTrackEvent } from '@edx/frontend-platform/analytics';
import { selectors } from '../../../../../../data/redux';
import analyticsEvents from './constants';

export const useTrackSocialSharingChange = ({ updateField }) => {
  const analytics = useSelector(selectors.app.analytics);
  const allowVideoSharing = useSelector(selectors.video.allowVideoSharing);
  return (event) => {
    sendTrackEvent(
      analyticsEvents.socialSharingSettingChanged,
      {
        ...analytics,
        value: event.target.checked,
      },
    );
    updateField({
      allowVideoSharing: {
        ...allowVideoSharing,
        value: event.target.checked,
      },
    });
  };
};

export default useTrackSocialSharingChange;
