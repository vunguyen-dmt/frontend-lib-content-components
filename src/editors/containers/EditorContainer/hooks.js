import { useState } from 'react';
import { useSelector } from 'react-redux';

import analyticsEvt from '../../data/constants/analyticsEvt';
import { RequestKeys } from '../../data/constants/requests';
import { selectors } from '../../data/redux';
import { StrictDict } from '../../utils';
import * as appHooks from '../../hooks';
import * as module from './hooks';

export const {
  clearSaveError,
  navigateCallback,
  nullMethod,
  saveBlock,
} = appHooks;

export const state = StrictDict({
  isCancelConfirmModalOpen: (val) => useState(val),
});

export const handleSaveClicked = ({ dispatch, getContent, validateEntry }) => {
  const destination = useSelector(selectors.app.returnUrl);
  const analytics = useSelector(selectors.app.analytics);

  return () => saveBlock({
    analytics,
    content: getContent({ dispatch }),
    destination,
    dispatch,
    validateEntry,
  });
};

export const cancelConfirmModalToggle = () => {
  const [isCancelConfirmOpen, setIsOpen] = module.state.isCancelConfirmModalOpen(false);
  return {
    isCancelConfirmOpen,
    openCancelConfirmModal: () => setIsOpen(true),
    closeCancelConfirmModal: () => setIsOpen(false),
  };
};

export const handleCancel = ({ onClose }) => {
  if (onClose) {
    return onClose;
  }
  return navigateCallback({
    destination: useSelector(selectors.app.returnUrl),
    analyticsEvent: analyticsEvt.editorCancelClick,
    analytics: useSelector(selectors.app.analytics),
  });
};

export const isInitialized = () => useSelector(selectors.app.isInitialized);

export const saveFailed = () => useSelector((rootState) => (
  selectors.requests.isFailed(rootState, { requestKey: RequestKeys.saveBlock })
));
