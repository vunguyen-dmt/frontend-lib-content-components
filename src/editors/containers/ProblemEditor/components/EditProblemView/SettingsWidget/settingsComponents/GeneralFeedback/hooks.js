import { useState, useEffect } from 'react';
import _ from 'lodash-es';
import messages from './messages';
import * as module from './hooks';

export const state = {
  // eslint-disable-next-line react-hooks/rules-of-hooks
  summary: (val) => useState(val),
};

export const generalFeedbackHooks = (generalFeedback, updateSettings) => {
  const [summary, setSummary] = module.state.summary({
    message: messages.noGeneralFeedbackSummary, values: {}, intl: true,
  });

  // eslint-disable-next-line react-hooks/rules-of-hooks
  useEffect(() => {
    if (_.isEmpty(generalFeedback)) {
      setSummary({ message: messages.noGeneralFeedbackSummary, values: {}, intl: true });
    } else {
      setSummary({
        message: generalFeedback,
        values: {},
        intl: false,
      });
    }
  }, [generalFeedback]);

  const handleChange = (event) => {
    updateSettings({ generalFeedback: event.target.value });
  };

  return {
    summary,
    handleChange,
  };
};
