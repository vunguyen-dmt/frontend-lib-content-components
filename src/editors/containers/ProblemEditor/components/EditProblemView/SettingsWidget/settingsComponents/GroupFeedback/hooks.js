import { useState, useEffect } from 'react';
import _ from 'lodash-es';
import messages from './messages';
import * as module from './hooks';

export const state = {
  summary: (val) => useState(val),
};

export const groupFeedbackCardHooks = (groupFeedbacks, updateSettings, answerslist) => {
  const [summary, setSummary] = module.state.summary({ message: messages.noGroupFeedbackSummary, values: {} });

  useEffect(() => {
    if (groupFeedbacks.length === 0) {
      setSummary({ message: messages.noGroupFeedbackSummary, values: {} });
    } else {
      const feedbacksInList = groupFeedbacks.map(({ answers, feedback }) => {
        const answerIDs = answerslist.map((a) => a.id);
        const answersString = answers.filter((value) => answerIDs.includes(value));
        return `${answersString} ${feedback}\n`;
      });
      setSummary({
        message: messages.groupFeedbackSummary,
        values: { groupFeedback: feedbacksInList },
      });
    }
  }, [groupFeedbacks, answerslist]);

  const handleAdd = () => {
    let newId = 0;
    if (!_.isEmpty(groupFeedbacks)) {
      newId = Math.max(...groupFeedbacks.map(feedback => feedback.id)) + 1;
    }
    const groupFeedback = { id: newId, answers: [], feedback: '' };
    const modifiedGroupFeedbacks = [...groupFeedbacks, groupFeedback];
    updateSettings({ groupFeedbackList: modifiedGroupFeedbacks });
  };

  return {
    summary,
    handleAdd,
  };
};

export const groupFeedbackRowHooks = ({ id, groupFeedbacks, updateSettings }) => {
  // Hooks for the answers associated with a groupfeedback
  const addSelectedAnswer = ({ value }) => {
    const oldGroupFeedback = groupFeedbacks.find(x => x.id === id);
    const newAnswers = [...oldGroupFeedback.answers, value];
    const newFeedback = { ...oldGroupFeedback, answers: newAnswers };
    const remainingFeedbacks = groupFeedbacks.filter((item) => (item.id !== id));
    const updatedFeedbackList = [newFeedback, ...remainingFeedbacks].sort((a, b) => a.id - b.id);

    updateSettings({ groupFeedbackList: updatedFeedbackList });
  };
  const removedSelectedAnswer = ({ value }) => {
    const oldGroupFeedback = groupFeedbacks.find(x => x.id === id);
    const newAnswers = oldGroupFeedback.answers.filter(item => item !== value);
    const newFeedback = { ...oldGroupFeedback, answers: newAnswers };
    const remainingFeedbacks = groupFeedbacks.filter((item) => (item.id !== id));
    const updatedFeedbackList = [newFeedback, ...remainingFeedbacks].sort((a, b) => a.id - b.id);

    updateSettings({ groupFeedbackList: updatedFeedbackList });
  };
  const handleAnswersSelectedChange = (event) => {
    const { checked, value } = event.target;
    if (checked) {
      addSelectedAnswer({ value });
    } else {
      removedSelectedAnswer({ value });
    }
  };

  // Delete Button
  const handleDelete = () => {
    const modifiedGroupFeedbacks = groupFeedbacks.filter((item) => (item.id !== id));
    updateSettings({ groupFeedbackList: modifiedGroupFeedbacks });
  };

  // Hooks for the feedback associated with a groupfeedback
  const handleFeedbackChange = (event) => {
    const { value } = event.target;
    const modifiedGroupFeedback = groupFeedbacks.map(groupFeedback => {
      if (groupFeedback.id === id) {
        return { ...groupFeedback, feedback: value };
      }
      return groupFeedback;
    });
    updateSettings({ groupFeedbackList: modifiedGroupFeedback });
  };

  return {
    handleAnswersSelectedChange, handleFeedbackChange, handleDelete,
  };
};
