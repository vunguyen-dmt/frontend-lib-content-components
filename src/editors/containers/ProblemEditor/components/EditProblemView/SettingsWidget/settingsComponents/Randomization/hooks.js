import { useState, useEffect } from 'react';
import { RandomizationTypes, RandomizationTypesKeys } from '../../../../../../../data/constants/problem';
import * as module from './hooks';

export const state = {
  // eslint-disable-next-line react-hooks/rules-of-hooks
  summary: (val) => useState(val),
};

export const useRandomizationSettingStatus = ({ randomization, updateSettings }) => {
  const [summary, setSummary] = module.state.summary({
    message: RandomizationTypes[RandomizationTypesKeys.NEVER],
    values: {},
  });
  useEffect(() => {
    setSummary({
      message: randomization ? RandomizationTypes[randomization] : RandomizationTypes[RandomizationTypesKeys.NEVER],
    });
  }, [randomization]);

  const handleChange = (event) => {
    updateSettings({ randomization: event.target.value });
  };
  return { summary, handleChange };
};

export default useRandomizationSettingStatus;
