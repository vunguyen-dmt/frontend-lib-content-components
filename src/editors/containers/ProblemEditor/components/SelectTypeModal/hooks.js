import { useEffect, useState } from 'react';
import {
  AdvanceProblemKeys, AdvanceProblems, ProblemTypeKeys, ProblemTypes,
} from '../../../../data/constants/problem';
import { StrictDict } from '../../../../utils';
import * as module from './hooks';
import { getDataFromOlx } from '../../../../data/redux/thunkActions/problem';

export const state = StrictDict({
  selected: (val) => useState(val),
});

export const selectHooks = () => {
  const [selected, setSelected] = module.state.selected(ProblemTypeKeys.SINGLESELECT);
  return {
    selected,
    setSelected,
  };
};

export const onSelect = ({ selected, updateField, setBlockTitle }) => () => {
  if (Object.values(AdvanceProblemKeys).includes(selected)) {
    updateField({ problemType: ProblemTypeKeys.ADVANCED, rawOLX: AdvanceProblems[selected].template });
    setBlockTitle(AdvanceProblems[selected].title);
  } else {
    const newOLX = ProblemTypes[selected].template;
    const { settings, ...newState } = getDataFromOlx({ rawOLX: newOLX, rawSettings: {} });
    updateField({ ...newState });
    setBlockTitle(ProblemTypes[selected].title);
  }
};

export const useArrowNav = (selected, setSelected) => {
  const detectKeyDown = (e) => {
    const problemTypeValues = Object.values(ProblemTypeKeys);
    switch (e.key) {
      case 'ArrowUp':
        if (problemTypeValues.includes(selected) && ProblemTypes[selected].prev) {
          setSelected(ProblemTypes[selected].prev);
          document.getElementById(ProblemTypes[selected].prev).focus();
        }
        break;
      case 'ArrowDown':
        if (problemTypeValues.includes(selected) && ProblemTypes[selected].next) {
          setSelected(ProblemTypes[selected].next);
          document.getElementById(ProblemTypes[selected].next).focus();
        }
        break;
      default:
    }
  };
  useEffect(() => {
    document.addEventListener('keydown', detectKeyDown, true);
    return () => {
      document.removeEventListener('keydown', detectKeyDown, true);
    };
  }, [selected, setSelected]);
};

export default {
  state,
  selectHooks,
  onSelect,
  useArrowNav,
};
