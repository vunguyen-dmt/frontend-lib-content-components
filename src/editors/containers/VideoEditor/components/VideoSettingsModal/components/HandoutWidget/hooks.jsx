import React from 'react';
import { useDispatch } from 'react-redux';
import { thunkActions } from '../../../../../../data/redux';
import * as module from './hooks';

export const state = {
  // eslint-disable-next-line react-hooks/rules-of-hooks
  showSizeError: (args) => React.useState(args),
};

export const parseHandoutName = ({ handout }) => {
  if (handout) {
    const handoutName = handout.slice(handout.lastIndexOf('@') + 1);
    return handoutName;
  }
  return 'None';
};

export const checkValidFileSize = ({
  file,
  onSizeFail,
}) => {
  // Check if the file size is greater than 20 MB, upload size limit
  if (file.size > 20000000) {
    onSizeFail();
    return false;
  }
  return true;
};

export const fileInput = ({ fileSizeError }) => {
  // eslint-disable-next-line react-hooks/rules-of-hooks
  const dispatch = useDispatch();
  // eslint-disable-next-line react-hooks/rules-of-hooks
  const ref = React.useRef();
  const click = () => ref.current.click();
  const addFile = (e) => {
    const file = e.target.files[0];
    if (file && module.checkValidFileSize({
      file,
      onSizeFail: () => {
        fileSizeError.set();
      },
    })) {
      dispatch(thunkActions.video.uploadHandout({
        file,
      }));
    }
  };
  return {
    click,
    addFile,
    ref,
  };
};

export const fileSizeError = () => {
  const [showSizeError, setShowSizeError] = module.state.showSizeError(false);
  return {
    fileSizeError: {
      show: showSizeError,
      set: () => setShowSizeError(true),
      dismiss: () => setShowSizeError(false),
    },
  };
};

export default { fileInput, fileSizeError, parseHandoutName };
