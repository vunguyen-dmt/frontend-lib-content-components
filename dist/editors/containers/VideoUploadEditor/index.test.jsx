import React from 'react';
import { IntlProvider } from '@edx/frontend-platform/i18n';
import { render, fireEvent, act } from '@testing-library/react';
import '@testing-library/jest-dom';
import VideoUploadEditor, { VideoUploader } from '.';
import * as hooks from './hooks';
import * as appHooks from '../../hooks';

const mockDispatch = jest.fn();
const mockOnUpload = jest.fn();

jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
}));
jest.mock('../../hooks', () => ({
  ...jest.requireActual('../../hooks'),
  navigateTo: jest.fn((args) => ({ navigateTo: args })),
}));

const defaultEditorProps = {
  intl: {},
};

const defaultUploaderProps = {
  onUpload: mockOnUpload,
  errorMessage: '',
  intl: {},
};

const renderEditorComponent = (props = defaultEditorProps) => render(
  <IntlProvider locale="en">
    <VideoUploadEditor {...props} />
  </IntlProvider>,
);

const renderUploaderComponent = (props = defaultUploaderProps) => render(
  <IntlProvider locale="en">
    <VideoUploader {...props} />
  </IntlProvider>,
);

describe('VideoUploadEditor', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders without errors', () => {
    const { container } = renderEditorComponent();
    expect(container).toMatchSnapshot();
  });

  it('updates the input field value when user types', () => {
    const { getByPlaceholderText } = renderEditorComponent();
    const input = getByPlaceholderText('Paste your video ID or URL');

    fireEvent.change(input, { target: { value: 'test value' } });
    expect(input.value).toBe('test value');
  });

  it('click on the save button', () => {
    const { getByPlaceholderText, getByTestId } = renderEditorComponent();
    const testValue = 'test vale';
    const input = getByPlaceholderText('Paste your video ID or URL');
    fireEvent.change(input, { target: { value: testValue } });
    const button = getByTestId('inputSaveButton');
    fireEvent.click(button);
    expect(appHooks.navigateTo).toHaveBeenCalled();
  });

  it('shows error message with unsupported files', async () => {
    const { getByTestId, findByText } = renderEditorComponent();
    const fileInput = getByTestId('fileInput');

    const unsupportedFile = new File(['(⌐□_□)'], 'unsupported.avi', { type: 'video/avi' });
    fireEvent.change(fileInput, { target: { files: [unsupportedFile] } });

    const errorMsg = await findByText('Video must be an MP4 or MOV file');
    expect(errorMsg).toBeInTheDocument();
  });

  it('calls uploadVideo with supported files', async () => {
    const uploadVideoSpy = jest.spyOn(hooks, 'uploadVideo');
    const { container } = renderEditorComponent();
    const dropzone = container.querySelector('.dropzone-middle');

    const supportedFile = new File(['(⌐□_□)'], 'supported.mp4', { type: 'video/mp4' });

    await act(async () => {
      fireEvent.drop(dropzone, {
        dataTransfer: {
          files: [supportedFile],
          types: ['Files'],
        },
      });
    });

    expect(uploadVideoSpy).toHaveBeenCalledWith(expect.objectContaining({
      dispatch: mockDispatch,
      supportedFiles: expect.arrayContaining([
        expect.objectContaining({
          name: supportedFile.name,
          type: supportedFile.type,
          size: supportedFile.size,
        }),
      ]),
    }));
  });
});

describe('VideoUploader', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders without errors', () => {
    const { container } = renderUploaderComponent();
    expect(container).toMatchSnapshot();
  });

  it('renders with an error message', () => {
    const errorMessage = 'Video must be an MP4 or MOV file';
    const { getByText } = renderUploaderComponent({ ...defaultUploaderProps, errorMessage });
    expect(getByText(errorMessage)).toBeInTheDocument();
  });

  it('calls the onUpload function when a supported file is dropped', async () => {
    const { container } = renderUploaderComponent();
    const dropzone = container.querySelector('.dropzone-middle');
    const file = new File(['(⌐□_□)'], 'video.mp4', { type: 'video/mp4' });

    await act(async () => {
      fireEvent.drop(dropzone, {
        dataTransfer: {
          files: [file],
          types: ['Files'],
        },
      });
    });

    expect(mockOnUpload).toHaveBeenCalledWith(file);
  });
});
