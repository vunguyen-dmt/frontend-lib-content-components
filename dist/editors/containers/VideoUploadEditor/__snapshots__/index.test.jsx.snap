// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoUploadEditor renders as expected with default behavior 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="d-flex marked-area flex-column p-3"
        >
          <div
            class="d-flex flex-column"
          >
            <div
              class="d-flex justify-content-end flex-row"
            >
              <button
                aria-label="Close"
                class="btn-icon btn-icon-primary btn-icon-md position-absolute mr-2 mt-2"
                type="button"
              >
                <span
                  class="btn-icon__icon-container"
                >
                  <span
                    class="pgn__icon btn-icon__icon"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="24"
                      role="img"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </span>
              </button>
            </div>
            <div
              class="pgn__dropzone"
              data-testid="dropzone-container"
              role="presentation"
              tabindex="0"
            >
              <input
                accept="video/*,.mp4,.mov"
                style="display: none;"
                tabindex="-1"
                type="file"
              />
              <div
                class="d-flex flex-column justify-content-around align-items-center w-100"
              >
                <div
                  class="d-flex flex-column"
                >
                  <div
                    class="justify-content-center align-self-center rounded-circle p-5"
                    style="background-color: rgb(242, 240, 239);"
                  >
                    <span
                      class="pgn__icon pgn__icon__lg text-muted"
                    >
                      <svg
                        aria-hidden="true"
                        fill="none"
                        focusable="false"
                        height="24"
                        role="img"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M5 10h4v6h6v-6h4l-7-7-7 7zm0 8v2h14v-2H5z"
                          fill="currentColor"
                        />
                      </svg>
                    </span>
                  </div>
                  <div
                    class="d-flex align-self-center justify-content-center flex-wrap flex-column pt-5"
                  >
                    <span
                      class="small"
                    >
                      Drag and drop video here or click to upload
                    </span>
                    <span
                      class="align-self-center"
                      style="font-size: 0.8rem;"
                    >
                      Upload MP4 or MOV files (5 GB max)
                    </span>
                  </div>
                  <div
                    class="x-small align-self-center justify-content-center mx-2 text-dark font-weight-normal"
                  >
                    OR
                  </div>
                  <div
                    class="zindex-9 video-id-prompt p-4"
                  >
                    <div
                      class="video-upload-input-group input-group"
                    >
                      <div
                        class="pgn__form-control-decorator-group"
                      >
                        <input
                          aria-describedby="basic-addon2"
                          aria-label="Paste your video ID or URL"
                          class="form-control"
                          placeholder="Paste your video ID or URL"
                          value=""
                        />
                      </div>
                      <div
                        class="light-300 justify-content-center align-self-center bg-light rounded-circle p-0 x-small url-submit-button"
                      >
                        <button
                          aria-label="Submit"
                          class="btn-icon btn-icon-primary btn-icon-inline text-muted"
                          type="button"
                        >
                          <span
                            class="btn-icon__icon-container"
                          >
                            <span
                              class="pgn__icon btn-icon__icon"
                            >
                              <svg
                                aria-hidden="true"
                                fill="none"
                                focusable="false"
                                height="24"
                                role="img"
                                viewBox="0 0 24 24"
                                width="24"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8-8-8Z"
                                  fill="currentColor"
                                />
                              </svg>
                            </span>
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="d-flex marked-area flex-column p-3"
      >
        <div
          class="d-flex flex-column"
        >
          <div
            class="d-flex justify-content-end flex-row"
          >
            <button
              aria-label="Close"
              class="btn-icon btn-icon-primary btn-icon-md position-absolute mr-2 mt-2"
              type="button"
            >
              <span
                class="btn-icon__icon-container"
              >
                <span
                  class="pgn__icon btn-icon__icon"
                >
                  <svg
                    aria-hidden="true"
                    fill="none"
                    focusable="false"
                    height="24"
                    role="img"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </div>
          <div
            class="pgn__dropzone"
            data-testid="dropzone-container"
            role="presentation"
            tabindex="0"
          >
            <input
              accept="video/*,.mp4,.mov"
              style="display: none;"
              tabindex="-1"
              type="file"
            />
            <div
              class="d-flex flex-column justify-content-around align-items-center w-100"
            >
              <div
                class="d-flex flex-column"
              >
                <div
                  class="justify-content-center align-self-center rounded-circle p-5"
                  style="background-color: rgb(242, 240, 239);"
                >
                  <span
                    class="pgn__icon pgn__icon__lg text-muted"
                  >
                    <svg
                      aria-hidden="true"
                      fill="none"
                      focusable="false"
                      height="24"
                      role="img"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M5 10h4v6h6v-6h4l-7-7-7 7zm0 8v2h14v-2H5z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </div>
                <div
                  class="d-flex align-self-center justify-content-center flex-wrap flex-column pt-5"
                >
                  <span
                    class="small"
                  >
                    Drag and drop video here or click to upload
                  </span>
                  <span
                    class="align-self-center"
                    style="font-size: 0.8rem;"
                  >
                    Upload MP4 or MOV files (5 GB max)
                  </span>
                </div>
                <div
                  class="x-small align-self-center justify-content-center mx-2 text-dark font-weight-normal"
                >
                  OR
                </div>
                <div
                  class="zindex-9 video-id-prompt p-4"
                >
                  <div
                    class="video-upload-input-group input-group"
                  >
                    <div
                      class="pgn__form-control-decorator-group"
                    >
                      <input
                        aria-describedby="basic-addon2"
                        aria-label="Paste your video ID or URL"
                        class="form-control"
                        placeholder="Paste your video ID or URL"
                        value=""
                      />
                    </div>
                    <div
                      class="light-300 justify-content-center align-self-center bg-light rounded-circle p-0 x-small url-submit-button"
                    >
                      <button
                        aria-label="Submit"
                        class="btn-icon btn-icon-primary btn-icon-inline text-muted"
                        type="button"
                      >
                        <span
                          class="btn-icon__icon-container"
                        >
                          <span
                            class="pgn__icon btn-icon__icon"
                          >
                            <svg
                              aria-hidden="true"
                              fill="none"
                              focusable="false"
                              height="24"
                              role="img"
                              viewBox="0 0 24 24"
                              width="24"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8-8-8Z"
                                fill="currentColor"
                              />
                            </svg>
                          </span>
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
